cmd/acr/purge.go

```diff
@@ -274,14 +279,39 @@ func getTagsToDelete(ctx context.Context,
 				tagsToDelete = append(tagsToDelete, tag)
 			}
 		}
-		// The lastTag is updated to keep the for loop going.
-		newLastTag = *tags[len(tags)-1].Name
+		newLastTag = getLastTagFromResponse(resultTags)
 		return &tagsToDelete, newLastTag, nil
 	}
 	// In case there are no more tags return empty string as lastTag so that the purgeTags function stops
 	return nil, "", nil
 }
 
+func getLastTagFromResponse(resultTags *acr.RepositoryTagsType) string {
```

Comment:
FYI, if the request tags are ordered by time, the returned `last` parameter may contains "&" character, which may cause unexpected issues during URL encoding and decoding.